pkg/kubelet/dockertools/exec.go

```diff
@@ -121,6 +126,14 @@ func (*NativeExecHandler) ExecInContainer(client DockerInterface, container *doc
 	if err != nil {
 		return err
 	}
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
+			// FIXME: we should kill the process in the container,
+			// but I couldn't find anything in the Docker API docs
+			// on how to do it with the *Exec APIs.
```

Comment:
I discovered it to be blocking in my terminal resizing PR. It didn't used to be. It's blocking because of https://github.com/kubernetes/kubernetes/blob/52ebd4ecf1970dae9b8ccb37a4941ec2e8dd9590/pkg/kubelet/dockertools/kube_docker_client.go#L427.

I assume that the for loop only ever runs once.

cc @Random-Liu